### DISCUSSION TITLE
Curl error - invalid grant

### DISCUSSION THREAD

**Discussion by drax16** [potmat/homebridge-google-nest-sdm#160] [Q&A]
First and foremost, thank you to potmat and everyone who has worked on this project. 

I've started from scratch several times, but once I get "close" to the end, I continuously get this error when trying to run the refreshToken command

> "error": "invalid_grant",
> "error_description": "Bad Request"

Someone online suggested running it via the homebridge terminal, but no luck.

This is the command (without my info) for the sake of specificity 

> curl -L -X POST 'https://www.googleapis.com/oauth2/v4/token?client_id=oauth2-client-id&client_secret=oauth2-client-secret&code=authorization-code&grant_type=authorization_code&redirect_uri=https://www.google.com'

**Comment by potmat** [comment]
Converted to discussion as this is not an issue with the plugin itself.

**Comment by potmat** [comment]
What did you actually put for "grant_type" note that it's supposed to be the literal value "authorization_code" not the authorization code you received in the previous step.

**Comment by mbdawson** [comment]
I was having the same error. You have to move quick after getting your auth_code, it does expire.
